core/src/test/java/org/infinispan/distexec/DistributedExecutorFailoverTest.java

```diff
@@ -124,4 +160,25 @@ public int maxFailoverAttempts() {
          return maxFailoverCount;
       }
    }
+
+   static class SameNodeTaskFailoverPolicy implements DistributedTaskFailoverPolicy {
+      private final int maxFailoverCount;
```

Comment:
I think those counters will be updated/read by different thread. Am I right? If so, please make them `volatille`.